File: eshop-integration/eAPI/v1/java/eAPI-v1-connector/src/main/java/cz/monetplus/mips/eapi/v1/connector/entity/ApiBase.java
package cz.monetplus.mips.eapi.v1.connector.entity;

import java.io.Serializable;

import com.google.gson.Gson;
import com.google.gson.GsonBuilder;

public abstract class ApiBase implements Serializable {

	private static final long serialVersionUID = -3825192932302805075L;
	
	public static final char SEP = '|'; 
	
	public static final int MERCHANT_ID_MAX_LEN = 10;
	public static final int ORDER_NO_MAX_LEN = 10;
	public static final int PAY_ID_MAX_LEN = 15;
	public static final int RETURNURL_MAX_LEN = 300;
	public static final int CART_ITEM_NAME_MAX_LEN = 20;
	public static final int CART_ITEM_DESC_MAX_LEN = 40;
	public static final int DESCRIPTION_MAX_LEN = 255;
	public static final int CUST_ID_MAX_LEN = 50;
	public static final int MERCHANTDATA_MAX_LEN = 255;
	public static final int SIGNATURE_MAX_LEN = 500;
	public static final int CARDNAME_MAX_LEN = 20;
	
	public static final String[] PAY_OPER_ALLOWED_VALUES = new String[] { "payment" };
	public static final String[] PAY_METHOD_ALLOWED_VALUES = new String[] { "card" };
	public static final String[] RETURN_METHOD_ALLOWED_VALUES = new String[] { "POST", "GET" };
	public static final String[] CURRENCY_ALLOWED_VALUES = new String[] { "CZK", "EUR", "USD", "GBP" };
	public static final String[] LANGUAGE_ALLOWED_VALUES = new String[] { "CZ", "EN", "DE", "SK" };
	
	public static final String DTTM_FORMAT = "YYYYMMddHHmmss";
	
	protected void add(StringBuffer sb, String value) {
		if (value != null) {
			sb.append(value).append(SEP);
		}
	}
	
	protected void add(StringBuffer sb, boolean value) {
		sb.append(value).append(SEP);
	}

	protected void add(StringBuffer sb, Integer value) {
		if (value != null) {
			sb.append(value).append(SEP);
		}
	}

	protected void add(StringBuffer sb, int value) {
		sb.append(value).append(SEP);
	}

	protected void add(StringBuffer sb, long value) {
		sb.append(value).append(SEP);
	}
	
	protected void deleteLast(StringBuffer sb) {
		if (SEP == sb.charAt(sb.length()-1)) {
			sb.deleteCharAt(sb.length()-1);
		}
	}

	public String toJson() {
		Gson gson = new GsonBuilder().create();
		return gson.toJson(this);
	}
	
}

File: Integration Examples/eAPI v1.8/dotNET/CsobGTWDemo/CsobGatewayClientExample/Communication/DataObjects/ApiBase.cs
using System.Text;

namespace CsobGatewayClientExample.Communication.DataObjects
{
    public class ApiBase
    {
        public const char SEP = '|';

        public const int MERCHANT_ID_MAX_LEN = 10;
        public const int ORDER_NO_MAX_LEN = 10;
        public const int PAY_ID_MAX_LEN = 15;
        public const int RETURNURL_MAX_LEN = 300;
        public const int CART_ITEM_NAME_MAX_LEN = 20;
        public const int CART_ITEM_DESC_MAX_LEN = 40;
        public const int DESCRIPTION_MAX_LEN = 255;
        public const int CUST_ID_MAX_LEN = 50;
        public const int MERCHANTDATA_MAX_LEN = 255;
        public const int SIGNATURE_MAX_LEN = 500;
        public const int CARDNAME_MAX_LEN = 20;

        public static readonly string[] PAY_OPER_ALLOWED_VALUES = new string[] { "payment" };
	    public static readonly string[] PAY_METHOD_ALLOWED_VALUES = new string[] { "card" };
	    public static readonly string[] RETURN_METHOD_ALLOWED_VALUES = new string[] { "POST", "GET" };
	    public static readonly string[] CURRENCY_ALLOWED_VALUES = new string[] { "CZK", "EUR", "USD", "GBP" };
	    public static readonly string[] LANGUAGE_ALLOWED_VALUES = new string[] { "CZ", "EN", "DE", "SK" };
	
	    public const string DTTM_FORMAT = "yyyyMMddHHmmss";
	
	    protected void Add(StringBuilder sb, string value)
        {
            if (value != null)
            {
                sb.Append(value).Append(SEP);
            }
        }

        protected void Add(StringBuilder sb, bool value)
        {
            sb.Append(value).Append(SEP);
        }

        protected void Add(StringBuilder sb, int? value)
        {
            if (value != null)
            {
                sb.Append(value).Append(SEP);
            }
        }

        protected void Add(StringBuilder sb, int value)
        {
            sb.Append(value).Append(SEP);
        }

        protected void Add(StringBuilder sb, long value)
        {
            sb.Append(value).Append(SEP);
        }

        protected void DeleteLast(StringBuilder sb)
        {
            if (SEP == sb[sb.Length - 1])
            {
                sb.Remove(sb.Length - 1, 1);
            }
        }
    }
}
